src/main/java/org/datadog/jmxfetch/util/ServiceCheckHelper.java

```diff
@@ -0,0 +1,12 @@
+package org.datadog.jmxfetch.util;
+
+import org.apache.commons.lang.StringUtils;
+
+public class ServiceCheckHelper {
+    /** Formats the service check prefix. */
```

Comment:
Let's add an explanation here about what this formatting does, and why, see explanation here: https://github.com/DataDog/jmxfetch/commit/0428c41ebf7a14404ae50928e3ecfc229701c042

This was, in hindsight, a poor solution to the technical problem at the time. As a way to start the transition to not using this formatting logic anymore, maybe it'd make sense to send both the formatted service check name and the unformatted one when your new option `check_prefix` is not specified, what do you think?

Reply:
> Formats the service check prefix name to enable having versionned .yaml
configuration files for the same checks (ex. : activemq/activemq_58)
while keeping on sending the same metric prefix. All numbers,
underscores and some other characters are simply removed from the prefix
if they are present.

I'm still not sure to understand why there was a need to strip those characters based on this comment ☝️ in order to have versioning (I don't have much context on this versioning concept).

> This was, in hindsight, a poor solution to the technical problem at the time. As a way to start the transition to not using this formatting logic anymore, maybe it'd make sense to send both the formatted service check name and the unformatted one when your new option check_prefix is not specified, what do you think?

Yes, make sense. So, if the check_name is `foo_bar`:

- If `check_prefix = foo`, we send 2 service checks `foo.can_connect` and `foobar.can_connect`
- If `check_prefix` is not set, we send 2 service checks `foo_can.connect` and `foobar.can_connect`
- We deprecate the current formatting: `foobar.can_connect`

Reply:
> I'm still not sure to understand why there was a need to strip those characters based on this comment ☝️ in order to have versioning (I don't have much context on this versioning concept).

at the time there were 2 JMXFetch integrations for `activemq`: one called `activemq` for older versions of activemq, the other called `activemq_58` for v5.8+ of activemq, see https://github.com/DataDog/dd-agent/tree/5.10.x/conf.d

And we still wanted both integrations to send the service check with the `activemq` prefix, hence this formatting function...

> So, if the check_name is foo_bar:
> - If check_prefix = foo, we send 2 service checks foo.can_connect and foobar.can_connect
> - If check_prefix is not set, we send 2 service checks foo_can.connect and foobar.can_connect
> - We deprecate the current formatting: foobar.can_connect

I agree overall. In the case of check_prefix = foo, it's up to you whether we still want to send `foobar.can_connect` (we should if you need that behavior for existing integrations where you'll start using `check_prefix`, otherwise we should only send `foo.can_connect`)

> But then we need a deprecation process for when we should remove the old behaviour ?

Yes, unclear how we'll do that. But at least JMXFetch will already be sending the "right" service check name.

Reply:
Just made changes related to this. Let me know if it's ok for you.